.travis.yml

```diff
@@ -27,36 +29,32 @@ addons:
     sources:
         - ubuntu-toolchain-r-test
     packages:
-        - g++-4.8
-        - ccache
         - wget
 
 before_install:
-    - mkdir -p ~/.local/bin
-    - export PATH=$HOME/.local/bin:$PATH
-    - travis_retry curl -L https://www.stackage.org/stack/linux-x86_64 | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
     - if [ "$TRAVIS_OS_NAME" == "linux" ]; then echo "Hardware:"; grep model\ name /proc/cpuinfo | uniq -c; free -m; fi
-    # linux prereqisite packages
-    - if [ "$TRAVIS_OS_NAME" == "linux" ]; then wget --no-check-certificate https://www.cmake.org/files/v3.2/cmake-3.2.3-Linux-x86_64.tar.gz; fi
-    - if [ "$TRAVIS_OS_NAME" == "linux" ]; then tar -xf cmake-3.2.3-Linux-x86_64.tar.gz; fi
-    - if [ "$TRAVIS_OS_NAME" == "linux" ]; then export PATH=$PWD/cmake-3.2.3-Linux-x86_64/bin:$PATH; fi
-    # set defaults
-    - if [ "$CXX" == "" ]; then echo "Setting default \$CXX"; export CXX="clang++"; fi
-    - if [ "$CC" == "" ]; then echo "Setting default \$CC"; export CC="clang"; fi
-    - if [ "$BOOST" == "" ]; then echo "Setting default \$BOOST"; export BOOST=1.63.0; fi
-    # pull down our boost binaries
-    - export BOOST_ARCHIVE=boost-$BOOST.tar.xz
-    - wget https://bondboostbinaries.blob.core.windows.net/bondboostbinaries/$BOOST_ARCHIVE -O /tmp/$BOOST_ARCHIVE
-    - pushd /; sudo tar -xf /tmp/$BOOST_ARCHIVE; popd
-    - export BOOST_ROOT=/opt/boost_`echo $BOOST | tr . _`
-    # ccache splits up your C{,XX}FLAGS and makes separate preprocessor and
-    # compiler calls. It includes a few flags that aren't used by one half of
-    # the process or the other. -Qunused-arguments prevents clang from failing
-    # the build because of this.
-    - export CXX="ccache $CXX -Qunused-arguments --system-header-prefix=boost/"
-    - export CC="ccache $CC -Qunused-arguments --system-header-prefix=boost/"
-    - echo "CXX=$CXX"
-    - echo "CC=$CC"
+
+    - if [ "$FLAVOR" == "cpp-core" ]; then export CPP_CMAKE_ARGS="-DBOND_SKIP_GBC_TESTS=TRUE -DBOND_ENABLE_GRPC=FALSE"; fi
+    - if [ "$FLAVOR" == "cpp-comm" ]; then export CPP_CMAKE_ARGS="-DBOND_ENABLE_COMM=TRUE -DBOND_SKIP_GBC_TESTS=TRUE -DBOND_SKIP_CORE_TESTS=TRUE -DBOND_ENABLE_GRPC=FALSE"; fi
+    - if [ "$FLAVOR" == "cpp-grpc" ]; then export CPP_CMAKE_ARGS="-DBOND_SKIP_GBC_TESTS=TRUE -DBOND_SKIP_CORE_TESTS=TRUE -DgRPC_ZLIB_PROVIDER=package"; fi
+
+    - if [ "$CPP_CMAKE_ARGS" != "" ]; then
+        travis_retry curl -L https://bondboostbinaries.blob.core.windows.net/bondboostbinaries/bond-xenial.tar.xz | docker load;
```

Comment:
I kiiiiind of want to know how long this takes. Travis vms are anemic enough that unxzing hundreds of megs seems like it could be significant. `| time docker load`?

Reply:
Updated.